### ISSUE TITLE
Android Adobe Analytics - NPE crash

### ISSUE THREAD

**Issue by alexvanyo** [Adobe-Marketing-Cloud/acp-sdks#391]
We are sometimes seeing a `NullPointerException` crash occurring on Android with Adobe Analytics version 1.2.4, with a stack trace of:
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List com.adobe.marketing.mobile.AnalyticsUnprocessedEvent.b()' on a null object reference
       at com.adobe.marketing.mobile.AnalyticsExtension.getSharedState(AnalyticsExtension.java:838)
       at com.adobe.marketing.mobile.AnalyticsExtension.processEvents(AnalyticsExtension.java:814)
       at com.adobe.marketing.mobile.AnalyticsExtension.handleAnalyticsRequestEvent(AnalyticsExtension.java:1228)
       at com.adobe.marketing.mobile.AnalyticsListenerGenericTrackRequestContent$1.run(AnalyticsListenerGenericTrackRequestContent.java:51)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:919)
```

**Comment by cacheung** [comment]
Thanks for stack trace, we will take a look.

Calise Cheung
Mobile Engineer
Adobe Experience Cloud

**Comment by praveek** [comment]
Hi @alexvanyo 
How frequent is this crash? Are you able to reproduce it locally?
Does you app use Launch rule to trigger analytics track action?

- Praveen

**Comment by alexvanyo** [comment]
It happens pretty infrequently, and I haven't been able to reproduce it locally yet.
I don't believe the app uses any Launch rules to trigger analytics tracking, we just call `trackAction` and `trackState` manually throughout the app.

**Comment by cacheung** [comment]
We also try to reproduce in house, but no luck yet.  If you see it again, can you copy the whole stack trace to us?

Best Regards,
Calise

**Comment by alexvanyo** [comment]
Sorry, I was incorrect before, it looks like we do have Launch rules set up for the app.

**Comment by praveek** [comment]
We fixed a crash around triggering analytics track state/action from launch rules which we plan to release soon. Can you share your app-id for me to check if that might be the cause of the crash ? 

-Praveen

**Comment by alexvanyo** [comment]
I'd prefer not to post the app id publicly, could we use an alternate form of communication?
My email is alex AT livefront.com

**Comment by emdobrin** [comment]
Hi @alexvanyo 
Can you please contact [Adobe Client Care](http://helpx.adobe.com/marketing-cloud/contact-support.html) and provide your config appid and the stack trace information that way?

Emilia Dobrin

**Comment by cacheung** [comment]
Hi Alex,
    Thanks for providing the app id.  We are still looking into it.

Calise

**Comment by cacheung** [comment]
As discussed over email we will update this issue once we release our next version.
Thanks,
Calise

**Comment by cacheung** [comment]
Hi @alexvanyo,
     We have this issue fixed in the latest Analytics Android 1.2.5 release.  

Thanks,
Calise

**Comment by alexvanyo** [comment]
Thank you everyone! We will update our app here to use 1.2.5 and re-add the launch rules we had removed as a workaround. I'll close this issue and reopen it or open a new one if we encounter problems again.
